splice_machine/src/test/java/com/splicemachine/derby/impl/sql/compile/NonCoveringIndexIT.java

```diff
@@ -0,0 +1,174 @@
+ */
+package com.splicemachine.derby.impl.sql.compile;
+
+import com.splicemachine.db.shared.common.reference.SQLState;
+import com.splicemachine.derby.test.framework.SpliceSchemaWatcher;
+import com.splicemachine.derby.test.framework.SpliceUnitTest;
+import com.splicemachine.derby.test.framework.SpliceWatcher;
+import com.splicemachine.derby.test.framework.TestConnection;
+import com.splicemachine.homeless.TestUtils;
+import com.splicemachine.test_tools.TableCreator;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.spark_project.guava.collect.Lists;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+
+import static com.splicemachine.test_tools.Rows.row;
+import static com.splicemachine.test_tools.Rows.rows;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by yxia on 2/1/18.
+ */
+@RunWith(Parameterized.class)
+public class NonCoveringIndexIT extends SpliceUnitTest {
+    private static final String SCHEMA = NonCoveringIndexIT.class.getSimpleName().toUpperCase();
+    private static SpliceWatcher spliceClassWatcher = new SpliceWatcher(SCHEMA);
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = Lists.newArrayListWithCapacity(8);
+        params.add(new Object[]{"NESTEDLOOP"});
+        params.add(new Object[]{"SORTMERGE"});
+        params.add(new Object[]{"BROADCAST"});
+        return params;
+    }
+
+    private String joinStrategy;
+
+    public NonCoveringIndexIT(String joinStrategy) {
+        this.joinStrategy = joinStrategy;
+    }
+
+    @ClassRule
+    public static SpliceSchemaWatcher spliceSchemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    @BeforeClass
+    public static void createSharedTables() throws Exception {
+        TestConnection connection=spliceClassWatcher.getOrCreateConnection();
+        new TableCreator(connection)
+                .withCreate("create table t1 (a1 int, b1 int, c1 int)")
+                .withInsert("insert into t1 values(?,?,?)")
+                .withRows(rows(row(1, 1, 1),
+                        row(2, 2, 2),
+                        row(3, 3, 3),
+                        row(4, 4, 4))).create();
+
+        new TableCreator(connection)
+                .withCreate("create table t2 (a2 int, b2 int, c2 int)")
+                .withIndex("create index ix_t2 on t2 (b2,c2)")
+                .withInsert("insert into t2 values(?,?,?)")
+                .withRows(rows(row(3, 3, 3),
+                        row(4, 4, 4),
+                        row(4, 5, 5))).create();
+    }
+
+    @Test
+    public void testJoinWithNonCoveringIndexInDT() throws Exception {
+        /* Q1 inner join */
+        String sql = format("select * from --splice-properties joinOrder=fixed\n" +
+                "t1,\n" +
```

Comment:
Just a style thing for me, I like my methods to test a single concept and if it fails to be clear what the issue is for the developer.  Seems like you are testing 4 items.

Reply:
Good tests by the way.

Reply:
Thanks @jleach4 . The reason I grouped 4 cases into the same function is that they are variations of the same issue.  I can put them in separate functions if it makes the test more clear. I'll push a change shortly.